File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode reverseList(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }
        int counter = 0;
        ArrayList<Integer> numbers = new ArrayList<Integer>();
        while (head != null) {
            numbers.add(0, head.val);
            head = head.next;
            counter++;
        }
        counter--;
        ListNode output = new ListNode(numbers.get(counter));
        counter--;
        while (counter >= 0) {
            ListNode temp = new ListNode(output.val, output.next);
            output.next = temp;
            output.val = numbers.get(counter);
            counter--;
        }
        return output;
    }
}

File: Data Structures & Algorithms/rotate-matrix/submission-0.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int size = matrix.size();
        cout << size << "   size\n";
        int size1 = size;
        for (int i = 0; i < size1; i++) {
            cout << i << "   i\n";
            for (int j = 0; j < size1 - 1 - i; j++) {
                cout << j << "   j\n";
                cout << size1 << "   size1\n";
                int temp = matrix[i][i+j];
                matrix[i][i+j] = matrix[size1-1-j][i];
                matrix[size1-1-j][i] = matrix[size1-1][size1-1-j];
                matrix[size1-1][size1-1-j] = matrix[i+j][size1-1];
                matrix[i+j][size1-1] = temp;
            }
            size1--;
        }
    }
};
